snap/snapenv/snapenv.go

```diff
@@ -63,6 +63,7 @@ func GetBasicSnapEnvVars(desc interface{}) []string {
 	return fillSnapEnvVars(desc, []string{
 		"SNAP={{.SnapPath}}",
 		"SNAP_DATA=/var{{.SnapPath}}",
+		"SNAP_SHARED_DATA=/var{{.SnapBasePath}}/shared",
```

Comment:
Per prior note, let's expose the App to the template and use the existing methods directly here instead of rebuilding the path locally.